Windows nightly debug build crashed

See error log [86572349576#step:15:887](https://github.com/solvcon/solvcon/actions/runs/29163391425/job/86572349576#step:15:887):

```
  Exit code 0xc0000409
  CMake Error: Generator: build tool execution failed, command was: D:/a/_temp/246192835/ninja.exe run_pilot_pytest
  ..\..\..\..\tests\test_pilot_2d.py::Save2DCanvasDialogTC::test_menu_action_is_registered 
  ninja: build stopped: subcommand failed.
```

It appeared the day before yesterday ([run 29116360412](https://github.com/solvcon/solvcon/actions/runs/29116360412)).  An analysis was done in https://github.com/yungyuc/solvcon/discussions/63 .

Happening again on 7/14: [86906682073#step:15:887](https://github.com/solvcon/solvcon/actions/runs/29276334383/job/86906682073#step:15:887).

Fixed by PR #1109.

----

After fix with PR #1109, Windows nightly debug still crashes ([87440609354#step:14:890](https://github.com/solvcon/solvcon/actions/runs/29441194731/job/87440609354#step:14:890)), but the error message changed:

```
  ..\..\..\..\tests\test_pilot.py::CallTipTC::test_seeded_callable_signature_and_first_paragraph PASSED [ 50%]
  Exit code 0xc0000374
  CMake Error: Generator: build tool execution failed, command was: D:/a/_temp/246192835/ninja.exe run_pilot_pytest
  ..\..\..\..\tests\test_pilot.py::CallTipTC::test_unknown_name_returns_empty PASSED [ 50%]
  ..\..\..\..\tests\test_pilot.py::WorkerTC::test_propagates_exceptions_through_the_future PASSED [ 50%]
  ..\..\..\..\tests\test_pilot.py::WorkerTC::test_returns_the_result PASSED [ 50%]
  ..\..\..\..\tests\test_pilot.py::WorkerTC::test_runs_off_the_calling_thread PASSED [ 50%]
  ..\..\..\..\tests\test_pilot.py::SetupProcessTC::test_broken_pilot_import_warns PASSED [ 50%]
  ..\..\..\..\tests\test_pilot.py::SetupProcessTC::test_namespace_includes_pilot PASSED [ 50%]
  ..\..\..\..\tests\test_pilot_2d.py::R2DWidgetWorldTC::test_add_2d_widget_created Windows fatal exception: code 0xc0000374
  
  Thread 0x000004d4 (most recent call first):
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\concurrent\futures\thread.py", line 90 in _worker
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\threading.py", line 1012 in run
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\threading.py", line 1075 in _bootstrap_inner
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\threading.py", line 1032 in _bootstrap
  
  Current thread 0x00000d58 (most recent call first):
    File "D:\a\solvcon\solvcon\tests\test_pilot_2d.py", line 157 in setUpClass
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\unittest.py", line 159 in unittest_setup_class_fixture
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\fixtures.py", line 998 in call_fixture_func
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\fixtures.py", line 1328 in pytest_fixture_setup
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_callers.py", line 121 in _multicall
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_manager.py", line 120 in _hookexec
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_hooks.py", line 512 in __call__
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\fixtures.py", line 1232 in execute
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\fixtures.py", line 726 in _get_active_fixturedef
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\fixtures.py", line 630 in getfixturevalue
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\fixtures.py", line 806 in _fillfixtures
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\python.py", line 1710 in setup
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\unittest.py", line 248 in setup
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\runner.py", line 536 in setup
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\runner.py", line 170 in pytest_runtest_setup
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_callers.py", line 121 in _multicall
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_manager.py", line 120 in _hookexec
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_hooks.py", line 512 in __call__
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\runner.py", line 250 in <lambda>
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\runner.py", line 361 in from_call
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\runner.py", line 249 in call_and_report
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\runner.py", line 132 in runtestprotocol
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\runner.py", line 118 in pytest_runtest_protocol
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_callers.py", line 121 in _multicall
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_manager.py", line 120 in _hookexec
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_hooks.py", line 512 in __call__
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\main.py", line 408 in pytest_runtestloop
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_callers.py", line 121 in _multicall
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_manager.py", line 120 in _hookexec
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_hooks.py", line 512 in __call__
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\main.py", line 384 in _main
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\main.py", line 330 in wrap_session
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\main.py", line 377 in pytest_cmdline_main
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_callers.py", line 121 in _multicall
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_manager.py", line 120 in _hookexec
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_hooks.py", line 512 in __call__
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\config\__init__.py", line 229 in _main
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\config\__init__.py", line 201 in main
    File "D:\a\solvcon\solvcon\solvcon\system.py", line 91 in _run_pytest
    File "D:\a\solvcon\solvcon\solvcon\system.py", line 125 in enter_main
  ninja: build stopped: subcommand failed.
```

Fixed by PR #1117.

----

Update https://github.com/solvcon/solvcon/issues/1058#issuecomment-4997562061

After the fix in PR #1117, the nightly windows debug run still show failure [87710751758#step:14:919](https://github.com/solvcon/solvcon/actions/runs/29524768857/job/87710751758#step:14:919):

```
  ..\..\..\..\tests\test_pilot_2d.py::R2DWidgetWorldTC::test_add_2d_widget_created Windows fatal exception: code 0xc0000374
  
  Thread 0x00000410 (most recent call first):
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\concurrent\futures\thread.py", line 90 in _worker
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\threading.py", line 1012 in run
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\threading.py", line 1075 in _bootstrap_inner
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\threading.py", line 1032 in _bootstrap
  
  Current thread 0x00000e90 (most recent call first):
    File "D:\a\solvcon\solvcon\tests\test_pilot_2d.py", line 157 in setUpClass
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\unittest.py", line 159 in unittest_setup_class_fixture
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\fixtures.py", line 998 in call_fixture_func
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\fixtures.py", line 1328 in pytest_fixture_setup
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_callers.py", line 121 in _multicall
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_manager.py", line 120 in _hookexec
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_hooks.py", line 512 in __call__
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\fixtures.py", line 1232 in execute
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\fixtures.py", line 726 in _get_active_fixturedef
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\fixtures.py", line 630 in getfixturevalue
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\fixtures.py", line 806 in _fillfixtures
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\python.py", line 1710 in setup
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\unittest.py", line 248 in setup
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\runner.py", line 536 in setup
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\runner.py", line 170 in pytest_runtest_setup
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_callers.py", line 121 in _multicall
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_manager.py", line 120 in _hookexec
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_hooks.py", line 512 in __call__
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\runner.py", line 250 in <lambda>
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\runner.py", line 361 in from_call
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\runner.py", line 249 in call_and_report
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\runner.py", line 132 in runtestprotocol
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\runner.py", line 118 in pytest_runtest_protocol
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_callers.py", line 121 in _multicall
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_manager.py", line 120 in _hookexec
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_hooks.py", line 512 in __call__
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\main.py", line 408 in pytest_runtestloop
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_callers.py", line 121 in _multicall
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_manager.py", line 120 in _hookexec
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_hooks.py", line 512 in __call__
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\main.py", line 384 in _main
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\main.py", line 330 in wrap_session
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\main.py", line 377 in pytest_cmdline_main
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_callers.py", line 121 in _multicall
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_manager.py", line 120 in _hookexec
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\pluggy\_hooks.py", line 512 in __call__
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\config\__init__.py", line 229 in _main
    File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\_pytest\config\__init__.py", line 201 in main
    File "D:\a\solvcon\solvcon\solvcon\system.py", line 91 in _run_pytest
    File "D:\a\solvcon\solvcon\solvcon\system.py", line 125 in enter_main
  ninja: build stopped: subcommand failed.
```

Fixed by PR #1134.

--- a/.claude/settings.json
+++ b/.claude/settings.json
@@ -16,6 +16,8 @@
       "Bash(make buildext:*)",
       "Bash(make install:*)",
       "Bash(cmake:*)",
+      "Bash(powershell.exe -NoProfile -File ./build.ps1:*)",
+      "Bash(powershell.exe -NoProfile -File build.ps1:*)",
       "Bash(ctest:*)",
       "Bash(pytest:*)",
       "Bash(python -m pytest:*)",

--- a/.gitattributes
+++ b/.gitattributes
@@ -1,0 +1,5 @@
+# Shell/agent tooling stays LF regardless of core.autocrlf.
+*.sh text eol=lf
+.claude/hooks/* text eol=lf
+.claude/statusline.sh text eol=lf
+.gitattributes text eol=lf

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,15 @@ project(solvcon)
 
 cmake_policy(SET CMP0148 OLD)
 
+if(MSVC)
+    # Use the release C runtime (/MD) in every config, matching the
+    # release-only Python, Qt6, and PySide6 we link. A Debug build would
+    # default to /MDd, whose different STL layout corrupts an std::string
+    # passed from a /MD dependency (e.g. QString::toStdString) and hangs the
+    # pilot at start-up. /Zi keeps the Debug build debuggable.
+    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreadedDLL")
+endif()
+
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake/")
 
 if(NOT SKIP_PYTHON_EXECUTABLE)

--- a/contrib/dependency/windows/build-scdv-windows.ps1
+++ b/contrib/dependency/windows/build-scdv-windows.ps1
@@ -822,12 +822,19 @@ function Build-Python {
     Copy-Item (Join-Path $ScdvUsrDir 'python.exe') `
         (Join-Path $ScdvUsrDir 'python3.exe') -Force
 
-    # A .pth (its "import" line runs at startup) adds bin/ to the DLL search
-    # path: PySide6 links this scdv's Qt DLLs in bin/, and Windows does not
-    # search PATH for a .pyd's dependencies (Python 3.8+).  Unix LD path analog.
+    # A .pth (its "import" line runs at startup) adds DLL search directories:
+    # Windows does not search PATH for a .pyd's dependencies (Python 3.8+), the
+    # Unix LD path analog.  bin/ holds this scdv's Qt DLLs; _solvcon.pyd also
+    # links pyside6.abi3.dll and shiboken6.abi3.dll directly, which live in
+    # their own site-packages dirs, so register those too (they load before
+    # PySide6's own __init__ runs its add_dll_directory).
     $pth = Join-Path $ScdvUsrDir 'Lib\site-packages\scdv_dll_directory.pth'
-    $pthline = 'import os, sys; _b = os.path.join(sys.prefix, ''bin''); ' +
-        'os.path.isdir(_b) and os.add_dll_directory(_b)'
+    $pthline = 'import os, sys, sysconfig; ' +
+        '_sp = sysconfig.get_paths().get(''purelib'') or ''''; ' +
+        '[os.add_dll_directory(_d) for _d in [' +
+        'os.path.join(sys.prefix, ''bin''), ' +
+        'os.path.join(_sp, ''PySide6''), ' +
+        'os.path.join(_sp, ''shiboken6'')] if os.path.isdir(_d)]'
     Set-Content -LiteralPath $pth -Value $pthline -Encoding ascii
 
     # The layout ships pip via --include-pip; bring setuptools/wheel/pip current

--- a/solvcon/pilot/_canvas_gui.py
+++ b/solvcon/pilot/_canvas_gui.py
@@ -58,6 +58,16 @@ class Save2DCanvasDialog(_gui_common.PilotFeature):
 
     def __init__(self, *args, **kw):
         super().__init__(*args, **kw)
+        # Build the dialog in run(), not here: parenting a QFileDialog to the
+        # main window before launch forces native window creation that aborts
+        # the Windows Debug CRT (exit 0xC0000409). Deferring keeps the parent
+        # (dialog stays window-modal) with no Qt window built before launch.
+        self._diag = None
+
+    def _ensure_dialog(self):
+        """Build the save dialog and its label controls on first use."""
+        if self._diag is not None:
+            return
         self._diag = QtWidgets.QFileDialog(self._mainWindow)
         self._diag.setOption(QtWidgets.QFileDialog.DontUseNativeDialog, True)
         self._diag.setAcceptMode(QtWidgets.QFileDialog.AcceptSave)
@@ -113,6 +123,7 @@ class Save2DCanvasDialog(_gui_common.PilotFeature):
             self._pycon.writeToHistory(
                 "Save 2D canvas: no focused 2D canvas\n")
             return False
+        self._ensure_dialog()
         self._init_label_controls(widget)
         self._on_filter_selected(self._diag.selectedNameFilter())
         self._diag.open(self, QtCore.SLOT("on_finished()"))
@@ -168,6 +179,8 @@ class Save2DCanvasDialog(_gui_common.PilotFeature):
             self._pycon.writeToHistory(
                 "Save 2D canvas: no focused 2D canvas\n")
             return False
+        # _export_overlay reads the dialog's label controls; build it first.
+        self._ensure_dialog()
         ok = widget.saveImage(path, self._export_overlay(widget))
         if ok:
             self._pycon.writeToHistory(f"Save 2D canvas: wrote {path}\n")

--- a/tests/test_pilot_2d.py
+++ b/tests/test_pilot_2d.py
@@ -514,11 +514,27 @@ class Save2DCanvasDialogTC(unittest.TestCase):
         # focused 2D subwindow, so clear the MDI before asserting the guard.
         mgr.mdiArea.closeAllSubWindows()
         feature = _canvas_gui.Save2DCanvasDialog(mgr=mgr)
-        opened = []
-        feature._diag.open = lambda *a, **k: opened.append(True)
         self.assertIsNone(mgr.currentR2DWidget())
         self.assertFalse(feature.run())
-        self.assertEqual(opened, [])
+        # The dialog is built lazily on first use, so a run with no focused
+        # canvas must not construct it at all.
+        self.assertIsNone(feature._diag)
+
+    def test_dialog_built_on_first_use(self):
+        """First use builds the dialog and its label controls; a second
+        build reuses the same dialog.
+        """
+        from PySide6 import QtWidgets
+        from solvcon.pilot import _canvas_gui
+        mgr = pilot.RManager.instance.setUp()
+        feature = _canvas_gui.Save2DCanvasDialog(mgr=mgr)
+        self.assertIsNone(feature._diag)
+        feature._ensure_dialog()
+        self.assertIsInstance(feature._diag, QtWidgets.QFileDialog)
+        self.assertIsNotNone(feature._labels_check)
+        built = feature._diag
+        feature._ensure_dialog()
+        self.assertIs(feature._diag, built)
 
     def test_save_current_reports_write_result(self):
         """Menu save path returns saveImage's bool and only writes on

